.github/workflows/ci.yml

```diff
@@ -84,10 +84,13 @@ jobs:
         with:
           path: target
           key: ${{ runner.os }}-build-rust_nightly-check-${{ hashFiles('**/Cargo.toml') }}
-      - uses: actions-rs/clippy-check@v1
+      - name: Clippy
+        uses: actions-rs/cargo@v1
```

Comment:
I think the reasonable alternative is using `actions-rs/clippy-check@v1` conditionally for non-PR pushes.
For ex:
```yaml
    uses: actions-rs/clippy-check@v1
    if: ${{ github.event_name != 'pull_request' }}
    with:
          token: ${{ secrets.GITHUB_TOKEN }}
          args: --all --all-features --all-targets -- -D warnings -A clippy::redundant_static_lifetimes
    uses: actions-rs/cargo@v1
    if: ${{ github.event_name == 'pull_request' }}
    command: clippy
          args: --all --all-features --all-targets -- -D warnings -A clippy::redundant_static_lifetimes
```

Reply:
I was thinking the same but the suggestion to use a permissions check through yet another Action seemed a bit overkill: https://github.com/actions-rs/clippy-check/issues/2#issuecomment-807878478.

Looking for `PR`s is a simple and elegant solution, let's go for that!

Reply:
@katyo I've dropped this change, `clippy-check` totally works without `GITHUB_TOKEN`, it just doesn't create a new task to pretty-print the outputs :tada:! https://github.com/Smithay/drm-rs/runs/2336361898

I probably didn't notice because of missing `-D warnings` and none of the failing crates were in the workspace yet.

Reply:
Odd though that GH itself doesn't parse all warnings to show as annotations on the front page, IIRC it used to do that before...